### ISSUE TITLE
Is  the Lua snippet `fun(ction)?` supposed to look like that?

### ISSUE THREAD

**Issue by mawkler** [mireq/luasnip-snippets#9]
The Lua snippet [`fun(ction)?`](https://github.com/mireq/luasnip-snippets/blob/849f7010eca25924d6bb207a6aeea3bb3218e217/lua/luasnip_snippets/lua.lua#L143-L147) is not expandable for me. Is it supposed to look like that?

**Comment by mireq** [comment]
@mawkler this triggers for:

- fun
- function

Here is [source](https://github.com/honza/vim-snippets/blob/aa52ad401bb1057c6c60d5adf37250bac79ce5fc/UltiSnips/lua.snippets#L20)

This is regex trigger and it has no list of alternatives to export to cmp. There are other triggers like lorem(number) which generates lorem ipsum with requested number of words. It's not practically possible to generate list of working alternatives to completion.

**Comment by mawkler** [comment]
@mireq So what you're saying is that the `fun(ction)` snippet does not work with LuaSnip? If that's the case, is it possible for luasnip-snippets to filter out those regex snippets so that they don't show up?

**Comment by mawkler** [comment]
@mireq Are you sure that it's a cmp_luasnip issue? I can't activate it with [your `minimal_lazy.lua` config](https://github.com/mireq/luasnip-snippets/blob/main/sample_configs/minimal_lazy.lua). If I type `func`, `funct`, `functi`, `functio` or `function` and press `<Tab>` with your config it doesn't work.

I can do `fun`, but that seems to be a different [`fun` snippet](https://github.com/mireq/luasnip-snippets/blob/875b8bd9ae9891c1163a4124b38a92d8a9521322/lua/luasnip_snippets/lua.lua#L51-L55).

**Comment by mireq** [comment]
@mawkler it should trigger only using `fun` and `function`. Rregex tells, that it should start with fun and then optionally can have  ction suffix (exactly this, not shorter). So only fun and function expands this snippet.

Correct shippet is [here](https://github.com/mireq/luasnip-snippets/blob/875b8bd9ae9891c1163a4124b38a92d8a9521322/lua/luasnip_snippets/lua.lua#L143-L147). It has higher priority.

**Comment by mawkler** [comment]
@mireq But, it doesn't trigger for `function`. When looking at [the source snippet from ultisnips](https://github.com/honza/vim-snippets/blob/aa52ad401bb1057c6c60d5adf37250bac79ce5fc/UltiSnips/lua.snippets#L20) I noticed that there's only a `fun(ction)` and no `fun` snippet, like lusnip-snippet has. Luasnip-snippets transpiles the ultisnips snippet to Lua, right? Does that mean that there should be `fun` and `function` snippet in LuaSnip, rather than a `fun` and a (broken) `fun(ction)` snippet?

**Comment by mireq** [comment]
@mawkler no, it's regular expression for both cases, so `fun(ction)` matches both, `fun` and `function`, but it don't display completion.
